Add Clerk Astro guide (#8908)
withastro/docs: src/content/docs/en/guides/authentication.mdx

- [`auth-astro` on GitHub](https://github.com/nowaythatworked/auth-astro?tab=readme-ov-file#auth-astro)
- [Auth.js documentation](https://authjs.dev/)

## Clerk

> Clerk is a complete suite of embeddable UIs, flexible APIs, and admin dashboards to authenticate and manage your users. An [official SDK](https://clerk.com/docs/references/astro/overview) for Astro is available.

Reviewer: ```suggestion
Clerk is a complete suite of embeddable UIs, flexible APIs, and admin dashboards to authenticate and manage your users. An [official Clerk SDK for Astro](https://clerk.com/docs/references/astro/overview) is available.
```
link text should be descriptive for assistive technology, so just moving the description into the link!

### Installation

Install `@clerk/astro` using the package manager of your choice.

<PackageManagerTabs>
  <Fragment slot="npm">
  ```shell
  npm install @clerk/astro
  ```
  </Fragment>
  <Fragment slot="pnpm">
  ```shell
  pnpm add @clerk/astro
  ```
  </Fragment>
  <Fragment slot="yarn">
  ```shell
  yarn add @clerk/astro
  ```
  </Fragment>
</PackageManagerTabs>

### Configuration

> Follow the ["Quick Start"](https://clerk.com/docs/quickstarts/astro) guide to set up Clerk integration and middleware in your Astro project.

Reviewer 2: ```suggestion
Follow the [Astro Quickstart](https://clerk.com/docs/quickstarts/astro) guide to set up Clerk integration and middleware in your Astro project.
```

> Follow the [Astro Quickstart](https://clerk.com/docs/quickstarts/astro) guide to set up Clerk integration and middleware in your Astro project.

Reviewer: ```suggestion
Follow [Clerk's own Astro Quickstart guide](https://clerk.com/docs/quickstarts/astro) to set up Clerk integration and middleware in your Astro project.
```
again, just better link text!

### Usage

> Protecting your pages in the client

Reviewer: ```suggestion
Clerk provides templating components that allow you to control the visibility of pages based on your visitor's authentication state. Show logged out users sign in button instead of content available to users who are logged in:
```

Author: I made a small change here. Let me know if that works!

> Clerk provides components that allow you to control the visibility of pages based on your user's authentication state. Show logged out users sign in button instead of content available to users who are logged in:

Reviewer: ```suggestion
Clerk provides components that allow you to control the visibility of pages based on your user's authentication state. Show logged out users a sign in button instead of the content available to users who are logged in:
```
I think the articles help the sentence here! So assuming we'd be on the same page, I'll just commit and get this published so we can start the weekend!  ;)

```astro title="src/pages/index.astro"
---
import Layout from 'src/layouts/Base.astro';
import { SignedIn, SignedOut, UserButton, SignInButton } from '@clerk/astro/components';
---

<Layout>
    <SignedIn>
        <UserButton />
    </SignedIn>
    <SignedOut>
        <SignInButton />
    </SignedOut>
</Layout>
```

> Protecting your routes in the server

Reviewer: ```suggestion
Clerk also allows you to protect routes on the server using middleware. Specify which routes are protected, and prompt unauthenticated users to sign in:
```

```ts title="src/middleware.ts"
import { clerkMiddleware, createRouteMatcher } from '@clerk/astro/server';

const isProtectedRoute = createRouteMatcher([
  '/dashboard(.*)',
  '/forum(.*)',
]);

export const onRequest = clerkMiddleware((auth, context) => {
  if (!auth().userId && isProtectedRoute(context.request)) {
    return auth().redirectToSignIn();
  }
});
```

### Next Steps

> - [`@clerk/astro` documentation](https://clerk.com/docs/references/astro/overview)

Reviewer: ```suggestion
- Read the [official `@clerk/astro` documentation](https://clerk.com/docs/references/astro/overview)
```

> - [Quickstart project](https://github.com/clerk/clerk-astro-quickstart)

Reviewer: ```suggestion
- Start from a template with the [Clerk + Astro Quickstart project](https://github.com/clerk/clerk-astro-quickstart)
```
